neqo-http3-server/src/main.rs

```diff
@@ -424,14 +438,44 @@ fn main() -> Result<(), io::Error> {
                             socket,
                         );
                         server.process_events(&args);
-                        process(
+                        if process(
+                            &mut **server,
+                            svr_timeout,
+                            event.token().0,
+                            None,
+                            &mut timer,
+                            socket,
+                        ) {
+                            active_servers.insert(event.token().0);
+                        }
+                    }
+                }
+            }
+            let curr_active = mem::replace(&mut active_servers, HashSet::new());
+            for inx in curr_active {
+                if let Some(socket) = sockets.get_mut(inx) {
+                    if let Some((ref mut server, svr_timeout)) =
+                        servers.get_mut(&socket.local_addr().unwrap())
+                    {
+                        let _ = process(
                             &mut **server,
                             svr_timeout,
                             event.token().0,
                             None,
                             &mut timer,
                             socket,
                         );
+                        server.process_events(&args);
+                        if process(
+                            &mut **server,
+                            svr_timeout,
+                            event.token().0,
+                            None,
+                            &mut timer,
+                            socket,
+                        ) {
+                            active_servers.insert(inx);
+                        }
```

Comment:
This block repeats here and above (Lines 432 to 451).  That means it can be extracted.